scalapact-core/src/main/scala/com/itv/scalapactcore/common/PactBrokerClient.scala

```diff
@@ -201,42 +201,94 @@ class PactBrokerClient(implicit
   def publishVerificationResults(
       pactVerifyResults: List[PactVerifyResult],
       brokerPublishData: BrokerPublishData,
+      providerVersionTags: List[String],
       pactBrokerAuthorization: Option[PactBrokerAuthorization],
       brokerClientTimeout: Option[Duration],
       sslContextName: Option[String]
   ): Unit = {
     val httpClient = httpClientBuilder.build(brokerClientTimeout.getOrElse(2.seconds), sslContextName, 2)
     pactVerifyResults.foreach { result =>
-      result.pact._links.flatMap(_.get("pb:publish-verification-results")).map(_.href) match {
-        case Some(link) =>
-          val success = !result.results.exists(_.result.isLeft)
-          val request = SimpleRequest(
-            link,
-            "",
-            HttpMethod.POST,
-            Map("Content-Type" -> "application/json; charset=UTF-8") ++ pactBrokerAuthorization.map(_.asHeader).toList,
-            body(brokerPublishData, success),
-            None
-          )
-          httpClient.doRequest(request) match {
-            case Right(response) =>
-              if (response.is2xx) {
-                PactLogger.message(
-                  s"Verification results published for provider '${result.pact.provider.name}' and consumer '${result.pact.consumer.name}'"
-                )
-              } else {
-                PactLogger.error(prettifyBrokerError("Publish verification results failed.", response))
-              }
-            case Left(err) => PactLogger.error(s"Unable to publish verification results: $err".red)
-          }
-        case None =>
-          PactLogger.error(
-            "Unable to publish verification results as there is no pb:publish-verification-results link".red
+      val publishedVerificationTags = result.pact._links
+        .flatMap(_.get("pb:provider"))
+        .map(_.href)
+        .map { providerUrl =>
+          publishVerificationResultTags(
+            httpClient,
+            providerUrl,
+            brokerPublishData.providerVersion,
+            providerVersionTags,
+            pactBrokerAuthorization
           )
+        }
+        .getOrElse {
+          PactLogger.error("Unable to publish verification results as there is no pb:provider link".red)
+          false
+        }
```

Comment:
IMO the provider should only be tagged if the verification results were successfully published, so this should come after

Reply:
That is what I assumed too, but Beth has informed me that we need to create the tags prior to verification in order for the "verification results published" webhooks to work properly.

> Beth (pactflow.io/Pact Broker/pact-ruby)
You can use the pact cli or a manual PUT as a workaround in the meantime. The tags need to be published BEFORE the results however, for the webhooks to kick off properly, so you may have some difficulties.

Reply:
maybe we can check this logic in another language's pact client implementation

Reply:
Oh cool, that is fine then